app/views/api/v2/hosts/main.json.rabl

```diff
@@ -37,6 +37,13 @@ node :hostgroup_name do |host|
   host.hostgroup.name if host.hostgroup.present?
 end
 
+node :creator do |host|
+  if host.creator_id.present?
+    user = User.find(host.creator_id)
+    "#{user.firstname} #{user.lastname}"
+  end
```

Comment:
I think this can be simplified to:
```suggestion
  host.creator&.fullname
```

However, It may be valuable to make it a hash where you can also provide the ID, rather than just the name.

Reply:
Hi @ekohl, I wasn't sure what you mean by making it a hash (to display in the UI?), but I added the `creator_id` param to the params we send in the response.
Please let me know if you meant something else.

Reply:
I meant that now you present the creator as a string, but the API is supposed to be generic. It's not just for the UI. So right now you return
```json
{
  "creator": "Some Name"
}
```

But what I was suggesting is:
```json
{
  "creator": {
    "id": 1234,
    "fullname": "Some Name"
  }
}
```

That way you can expand it in the future. You can also create a link to the user or search for all hosts created by the user (searching using names is a bad idea here, ID is much more reliable).

And if there is no creator, it should be:
```json
{
  "creator": null
}
```

Reply:
I got what you mean. 
But for example for the host's owner, we use `owner_id` and `owner_name` for the API and only display the `owner_name` in the UI. Isn't that enough? Or is it better to present it as a hash for future uses (like expanding it)?

Reply:
I think it's a good question. In retrospect (and if you're design an API v3) I'd change owner to a hash, just like I suggested for creator. In that case I'd even add a field `href` with `/api/v3/users/:id` as the value so clients don't need to guess the URLs.

However, an API v3 is without doubt out of scope here. The question is then what we should do now. A few options:

* remain consistent and add `creator_name` (or `creator_fullname`) and `creator_id`.
* Implement what I suggested above
* Do both: add `creator`, `creator_name` and `creator_id`

This is something where I'd get more feedback from others. You could open a post with a poll on https://community.theforeman.org/c/development/9 or even as an RFC. Others, such as the authors of https://github.com/theforeman/foreman-ansible-modules (large user of the API), can weigh in.

Reply:
Thank you Ewoud for all the help and for that clear and detailed answer!

Reply:
I opened this RFC [post](https://community.theforeman.org/t/rfc-adding-hosts-creator-field/31231) to see what others think about it.

Reply:
From the responses on the post, it seems that the first option (remain consistent and send two separate fields) is acceptable.
Please let me know if anything else is needed!